src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -6,9 +6,31 @@
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LENGTH_DATA = 6;
 
     @Override
     public User register(User user) {
-        return null;
+        if (checkLoginIfPresent(user) != null ) {
+            throw new InvalidDataException("User with this login has been already registered");
+        } else if (user.getAge() == null) {
+            throw new InvalidDataException("Age must not be null");
+        } else if (user.getLogin() == null) {
```

Comment:
The order of validations is not consistent with the checklist recommendations. It's better to check for null values of 'login', 'password', and 'age' before any other validations, starting with 'login' as it's the first parameter to be used.